src/platform/interpreter/constants.ts

```diff
@@ -6,397 +6,222 @@ export const JupyterKernelStartFailureOverrideReservedName =
+// And then check if we can import these as modules.
+// E.g. `ArithmeticError` is an allowed file/folder name, as that cannot be imported, hence the assumption is that
+// creating a file named `ArithmeticError.py` will not interfere with the startup of the kernel (tested and validated).
+// This way we only focus on files that we know can be imported by Python.
+// Unfortunately we don't know what the full list of files is, as users could have multiple packages installed, that could in turn pull in different dependencies
+// that in turn import these built-in modules.
 export const BuiltInModules = [
-    'ArithmeticError',
-    'AssertionError',
-    'AttributeError',
-    'BaseException',
-    'BlockingIOError',
-    'BrokenPipeError',
-    'BufferError',
-    'BytesWarning',
-    'ChildProcessError',
-    'ConnectionAbortedError',
-    'ConnectionError',
-    'ConnectionRefusedError',
-    'ConnectionResetError',
-    'DeprecationWarning',
-    'EOFError',
-    'Ellipsis',
-    'EnvironmentError',
-    'Exception',
-    'False',
-    'FileExistsError',
-    'FileNotFoundError',
-    'FloatingPointError',
-    'FutureWarning',
-    'GeneratorExit',
-    'IOError',
-    'ImportError',
-    'ImportWarning',
-    'IndentationError',
-    'IndexError',
-    'InterruptedError',
-    'IsADirectoryError',
-    'KeyError',
-    'KeyboardInterrupt',
-    'LookupError',
-    'MemoryError',
-    'ModuleNotFoundError',
-    'NameError',
-    'None',
-    'NotADirectoryError',
-    'NotImplemented',
-    'NotImplementedError',
-    'OSError',
-    'OverflowError',
-    'PendingDeprecationWarning',
-    'PermissionError',
-    'ProcessLookupError',
-    'RecursionError',
-    'ReferenceError',
-    'ResourceWarning',
-    'RuntimeError',
-    'RuntimeWarning',
-    'StopAsyncIteration',
-    'StopIteration',
-    'SyntaxError',
-    'SyntaxWarning',
-    'SystemError',
-    'SystemExit',
-    'TabError',
-    'TimeoutError',
-    'True',
-    'TypeError',
-    'UnboundLocalError',
-    'UnicodeDecodeError',
-    'UnicodeEncodeError',
-    'UnicodeError',
-    'UnicodeTranslateError',
-    'UnicodeWarning',
-    'UserWarning',
-    'ValueError',
-    'Warning',
-    'ZeroDivisionError',
-    '_',
-    '__build_class__',
-    '__debug__',
-    '__doc__',
-    '__future__',
-    '__import__',
-    '__loader__',
-    '__main__',
-    '__name__',
-    '__package__',
-    '__spec__',
-    '_aix_support',
-    '_bootlocale',
-    '_bootsubprocess',
-    '_collections_abc',
-    '_compat_pickle',
-    '_compression',
-    '_markupbase',
-    '_osx_support',
-    '_py_abc',
-    '_pydecimal',
-    '_pyio',
-    '_sitebuiltins',
-    '_strptime',
-    '_sysconfigdata_x86_64_conda_cos6_linux_gnu',
-    '_sysconfigdata_x86_64_conda_linux_gnu',
-    '_thread',
-    '_threading_local',
-    '_weakrefset',
-    'abc',
-    'abs',
-    'aifc',
-    'all',
-    'antigravity',
-    'any',
-    'argparse',
-    'array',
-    'ascii',
+    // 'abc',
```

Comment:
Shouldn't we put these back now that the diagnostic is turned off? Otherwise we're guessing here that these modules are okay to override in a kernel situation.

Reply:
Sure thing

Reply:
& agreed